web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/CdsProfileFetcher.java

```diff
@@ -37,16 +38,16 @@
 
 import java.io.IOException;
 import java.net.MalformedURLException;
+import java.net.URI;
 import java.net.URL;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 
-public class CdsProfileFetcher implements AppProfileFetcher {
+public class CdsProfileFetcher implements AppProfileFetcher, ApplicationIdResolver {
```

Comment:
what was the reason for not passing in config to constructor, and instead passing it in fetch method?

Reply:
This class is only used as a member of InstrumentationKeyResolver which is an enum singleton and therefore can't take parameters to its constructor.

IMO we should try to avoid singletons as much as possible.

Reply:
make sense